tests/integration/src/test/java/org/apache/pulsar/tests/integration/io/sinks/KinesisSinkTester.java

```diff
@@ -64,16 +63,11 @@ public void prepareSink() throws Exception {
         final LocalStackContainer localStackContainer = getServiceContainer();
         final URI endpointOverride = localStackContainer.getEndpointOverride(LocalStackContainer.Service.KINESIS);
         sinkConfig.put("awsEndpoint", NAME);
-        sinkConfig.put("awsEndpointPort", endpointOverride.getPort());
+        sinkConfig.put("awsEndpointPort", LOCALSTACK_SERVICE_PORT);
```

Comment:
Is this change necessary?

Reply:
it is the core of the problem :-)

Reply:
yes it is.
We were using the port forwarded to the host. But the sink is not on the host but in a container so it must use the internal container port.

Reply:
ok makes sense. Thanks for explaining.